entry/src/entry.rs

```diff
@@ -111,15 +111,22 @@ pub struct Api<'a> {
 /// field points to Transactions that took place shortly before `hash` was generated.
 ///
 /// If you multiply `num_hashes` by the amount of time it takes to generate a new hash, you
-/// get a duration estimate since the last Entry. Since processing power increases
+/// get a duration estimate since the last `Entry`. Since processing power increases
 /// over time, one should expect the duration `num_hashes` represents to decrease proportionally.
 /// An upper bound on Duration can be estimated by assuming each hash was generated by the
 /// world's fastest processor at the time the entry was recorded. Or said another way, it
 /// is physically not possible for a shorter duration to have occurred if one assumes the
 /// hash was computed by the world's fastest processor at that time. The hash chain is both
 /// a Verifiable Delay Function (VDF) and a Proof of Work (not to be confused with Proof of
 /// Work consensus!)
-
+///
+/// `Entry` contains `transactions` that are executable in parallel.  So an `Entry` is a unit of
+/// parallelism at the chain level.  All transactions in the `transactions` field have to follow the
+/// read/write locking restrictions with regard to the accounts they reference.  An account can be
+/// either written by a single transaction, or read by one or more transactions in the set.
+///
+/// This enforcement is done via a call to `solana_runtime::accounts::Accounts::lock_accounts()`
+/// with the `txs` argument holding all the `transactions` in the `Entry`.
```

Comment:
I like the addition of this piece as it adds a paper-trail for the reader to follow. Do you think it would be worthwhile to call out that the enforcement is done both my leaders constructing blocks as well as by validators replaying the blocks? I'm on the fence about anything additional in this regard, so fine with you adding a piece of leaving as-is

Reply:
All protocol level rules, I would assume, are enforced by everyone.
So, it seems to be the default, no?
Is there a rule that is only checked by the leader and not by the validators?
Or the other way around?

Reply:
> All protocol level rules, I would assume, are enforced by everyone. So, it seems to be the default, no?

That's true / fair

> Is there a rule that is only checked by the leader and not by the validators?

There is not, nor is there a rule that is checked by validators but not the leader. As you already called out, everyone (leader and validators) have to enforce the same rules in order to have a chance of agreeing on state after a given block.

> So, it seems to be the default, no?

So yes, it is the default; my comment was to provide extra context / explanation for the reader. Like I said, what you had was fine too so you can disregard my addition comment